examples/amp-story/fake-ad.html

```diff
@@ -0,0 +1,206 @@
+<!doctype html>
+<html amp lang="en">
+  <head>
+    <meta charset="utf-8">
+    <script async src="https://cdn.ampproject.org/v0.js"></script>
+    <script async custom-element="amp-video" src="https://cdn.ampproject.org/v0/amp-video-0.1.js"></script>
+    <script async custom-element="amp-audio" src="https://cdn.ampproject.org/v0/amp-audio-0.1.js"></script>
+    <script async custom-element="amp-story" src="https://cdn.ampproject.org/v0/amp-story-1.0.js"></script>
+    <script async custom-element="amp-story-auto-ads" src="https://cdn.ampproject.org/v0/amp-story-auto-ads-0.1.js"></script>
+    <title>My Story</title>
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+    <link rel="canonical" href="helloworld.html">
+    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+    <style amp-custom>
+      body {
+        font-family: 'Roboto', sans-serif;
+      }
+      amp-story-page {
+        background: white;
+      }
+      h1 {
+        font-size: 2.875em;
+        font-weight: normal;
+        line-height: 1.174;
+        text-transform: uppercase;
+      }
+    </style>
+  </head>
+
+  <body>
+    <amp-story
+      title="Amp-story-doubleclick"
+      publisher="The AMP Team"
+      publisher-logo-src="https://example.com/logo/1x1.png"
+      poster-portrait-src="https://example.com/my-story/poster/3x4.jpg"
+      poster-square-src="https://example.com/my-story/poster/1x1.jpg"
+      poster-landscape-src="https://example.com/my-story/poster/4x3.jpg"
+      standalone>
+
+        <amp-story-auto-ads>
+            <script type="application/json">
+              {
+                "ad-attributes": {
+                  "type": "fake",
+                  "src": "/examples/amp-story/ads/app-install.html",
+                  "id": "i-amphtml-demo-fake",
```

Comment:
Does amp-story-auto-ads check the validness of the id?

Reply:
amp-story-auto ads just creates an `<amp-ad type="fake" id="i-amphtml..">`. There is no additional logic around checking the id, so it should behave the same as a normal "fake" ad. Are you worried about the doc being invalid?

Reply:
After thinking about this more, I think I understand. The doc will be valid when it shouldn't be. I don't have a great solution. We could have the runtime not allow this in certain contexts, but I'm not sure what the signal would be. What do you think?

Reply:
To allowing fake ad specifically we can have some kind of workaround. For example, one must put `id=i-amphtml-...` to the `<amp-story-auto-ads>` component. So the fake ad only loads for invalid AMP page. 

However i think this is related to #20003. Allowing developers to randomly assign attribute of the inserted ad in general sounds very unsafe to me. I can easily inject something like `i-amphtml-ghost` without validator catching that.

Reply:
This is a good point. I agree a whitelist sounds good, but needs to be fleshed out. For instance `id` is allowed, but certain values are not. I will make some comments on the issue. 

To solve this problem I think that having an invalid signal on the amp-story-auto-ads tag is a good idea. I will refactor.

Reply:
SG. Thank you

Reply:
a different issue: are we inserting multiple `amp-ad` with the same ID?

Reply:
yes, is this not ok for testing?

Reply:
changed this around a bit so auto ads injects a unique id. Pubs just add the attribute to the auto-ads tag itself.